src/macaron/slsa_analyzer/checks/build_as_code_check.py

```diff
@@ -130,164 +86,98 @@ def run_check(self, ctx: AnalyzeContext, check_result: CheckResult) -> CheckResu
-                                break
-                            check_result["justification"].append(
-                                f"The target repository uses build tool {build_tool.name}"
-                                + f" in {ci_service.name} using {deploy_kw} to deploy."
+
+                            predicate["metadata"]["buildInvocationId"] = deploy_method.html_url
+                            predicate["builder"]["id"] = deploy_method.source_link
+                            predicate["invocation"]["configSource"]["entryPoint"] = deploy_method.trigger_link
+
+                            if highest_certainty == "deploy_kws":
+                                predicate["builder"]["id"] = deploy_method.config_name
+                                predicate["invocation"]["configSource"]["entryPoint"] = deploy_method.config_name
+
+                        check_result["result_tables"] = [
+                            BuildAsCodeTable(
+                                build_tool_name=build_tool.name,
+                                ci_service_name=ci_service.name,
+                                build_trigger=deploy_method.trigger_link,
+                                deploy_command=deploy_method.deploy_cmd,
+                                build_status_url=deploy_method.html_url,
+                                confidence_score=confidence_score,
                             )
-                            if ctx.dynamic_data["is_inferred_prov"] and ci_info["provenances"]:
-                                predicate = ci_info["provenances"][0]["predicate"]
-                                predicate["buildType"] = f"Custom {ci_service.name}"
-                                predicate["builder"]["id"] = config_name
-                                predicate["invocation"]["configSource"][
-                                    "uri"
-                                ] = f"{ctx.remote_path}@refs/heads/{ctx.branch_name}"
-                                predicate["invocation"]["configSource"]["digest"]["sha1"] = ctx.commit_sha
-                                predicate["invocation"]["configSource"]["entryPoint"] = config_name
-                            check_result["result_tables"] = [
-                                BuildAsCodeTable(
-                                    build_tool_name=build_tool.name,
-                                    ci_service_name=ci_service.name,
-                                    deploy_command=deploy_kw,
-                                )
-                            ]
-                            return CheckResultType.PASSED
+                        ]
+
+                # TODO: compile all justifications
+                # check_result["justification"].append()
+
+                # TODO: Investigate using proofs
+
+                # Check whether the confidence score is greater than the minimum threshold for this check.
+                if confidence_score >= self.confidence_score_threshold:
```

Comment:
At this line, if a CI service has a `confidence_score` larger than the threshold, but smaller than other CI services, wouldn't the check return the result for the smaller `confidence_score`?

Reply:
True. I can update this implementation to store all of the required information (i.e. deploy commands) for each `ci_service` and then populate the `BuilasAsCodeTable` and `AnalyzeContext` with whichever `ci_service` has the highest `confidence_score`?

Reply:
Sure, however I don't understand why `AnalyzeContext` needs to be updated?

Reply:
I just meant where the `ci_info` information is updated: https://github.com/oracle-samples/macaron/blob/add-problog-rule-inference-for-build-as-code/src/macaron/slsa_analyzer/checks/build_as_code_check.py#L145-L159.

Reply:
Oh, that's the inferred provenance representation. We don't update `ci_info` itself.